src/Ouroboros/Network/MonadClass/MonadSTM.hs

```diff
@@ -184,3 +187,18 @@ isEmptyTMVarDefault (TMVar t) = do
     Nothing -> return True
     Just _  -> return False
 
+class MonadSTM m => MonadTBQueue m where
+    type TBQueue m :: * -> *
```

Comment:
Can we follow the same pattern as the `TMVar` case? It doesn't need to be a separate class. Any instance that supports `TVar` can support the derived APIs using the defaults.